websites/R/Raven Code/presence.ts

```diff
@@ -0,0 +1,106 @@
+const presence = new Presence({ clientId: "937015924425367643" }),
+	startTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			details: "Geziniyor",
+			startTimestamp,
+		},
+		{ pathname, search } = document.location,
+		searchParams = new URLSearchParams(search);
+
+	switch (pathname) {
+		case "/":
+			presenceData.details = "Anasayfayı görüntülüyor";
+			break;
+		case "/yetkililer":
+			presenceData.details = "Yetkililer sayfasını görüntülüyor";
+			break;
+		case "/share":
+			presenceData.details = "Kod paylaşıyor";
+			break;
+		case "/tools":
+			presenceData.details = "Araçlar sayfasını görüntülüyor";
+			break;
+
+		case "/profile":
+			presenceData.details = "Profil görüntüleniyor:";
+			presenceData.state = document.querySelector(".user-info").textContent;
+			presenceData.largeImageKey =
+				document.querySelector<HTMLImageElement>("img.avatar").src;
+			break;
+
+		case "/codes":
+			presenceData.details = "Kodlar sayfasını görüntülüyor";
+			if (searchParams.get("filter") !== "all") {
+				presenceData.state = `Arama: ${decodeURIComponent(
+					searchParams.get("filter")
+				)} — Sayfa ${searchParams.get("page")}`;
+			} else presenceData.state = `Sayfa ${searchParams.get("page")}`;
+	}
+
+	if (pathname.startsWith("/code/")) {
+		presenceData.details = "Kod görüntüleniyor:";
+		presenceData.state = document.title.split("— ")[1];
+		const languages = [
+			{
+				page: "discord.js",
+				key: "https://avatars.githubusercontent.com/u/26492485?s=200&v=4",
+				text: "Discord.js",
```

Comment:
Can’t you get this data from the website?

Reply:
I can upload it to Imgur?

Reply:
I asked if you could get this data from the website

Reply:
Yes i can

Reply:
Is it better now?

Reply:
This isn’t what I was asking. I was asking if you could get the data in this array dynamically instead of hard coding it

Reply:
I prefer it that way, but yes, I can do.

Reply:
Getting it from the website dynamically would make it more future proof so it is preferable

Reply:
well I'll do it

Reply:
okay now?